src/my_project/tagging/lexer.py

```
@@ -84,6 +79,14 @@ def __init__(self, string: str) -> None:
         self.sbrace_depth: int = 0
         self.items: list[Item] = []
 
+    def format_items(self) -> str:
+        """Returns all lexer items in a readable format for debugging."""
+        lines = [f"Lexer items ({len(self.items)}):"]
+        for item in self.items:
+            lines.append(item.__repr__())
```

Ruff (0.16.1)

[warning] Use `list.extend` to create a transformed list

(PERF401)